### PR TITLE
Add workflow files to deploy `/api_inference`

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface-internal/api-inference/pull/2024.

This PR aims at replacing https://huggingface.co/docs/api-inference/index with new docs from https://github.com/huggingface/hub-docs/pull/1379.

Documentation lives in https://github.com/huggingface/hub-docs/tree/main/docs/api-inference and is semi-generated (see [scripts](https://github.com/huggingface/hub-docs/tree/main/scripts/api-inference)) based on [jsonschema specs](https://github.com/huggingface/huggingface.js/tree/main/packages/tasks/src/tasks) + [hf.co/tasks](https://hf.co/tasks) + Hub [code snippets](https://huggingface.co/black-forest-labs/FLUX.1-dev?inference_api=true).

For now the generation script has to be run locally but the goal is to automatically trigger it in the CI (to be done in a follow-up PR).

### CODE REVIEW

**Review by osanseviero — APPROVED** [review]
LGTM!

File: .github/workflows/api_inference_upload_pr_documentation.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Upload Inference API PR Documentation
```
Comment (mishig25) [inline .github/workflows/api_inference_upload_pr_documentation.yml:1]: https://github.com/huggingface/hub-docs/blob/main/.github/workflows/build_pr_documentation.yml for API-inference is missing

Comment (Wauplin) [reply .github/workflows/api_inference_upload_pr_documentation.yml:1]: thanks for spotting, it was in my untracked files ^^

**Review by mishig25 — APPROVED** [review]

**Comment by Wauplin** [comment]
Time to ship now that https://github.com/huggingface-internal/api-inference/pull/2024 is merged! Let's :crossed_fingers: everything runs smoothly :D

**Pull request merged**